### ISSUE TITLE
Bug - Tree view: separate selection/expansion example cannot operate via keyboard

### ISSUE THREAD

**Issue by thatblindgeye** [patternfly/patternfly-react#7759]
**Describe the problem**
Followup to #7714

The new example introduced in the above PR will need to have keyboard handling fixed, as currently you are unable to properly select tree items nor expand them. You can get into the toggle for the first item in the example to expand/collapse it, but keyboard focus remains stuck there unless you tab out of the entire tree.

Currently, Right Arrow will select a tree, and sometimes Left Arrow will as well.

**How do you reproduce the problem?**
Go to the [with separate selection and expansion tree view example](https://patternfly-react-pr-7714.surge.sh/components/tree-view#with-separate-selection-and-expansion) from the above PR's preview build and navigate/operate via keyboard. 

**Expected behavior**
Be able to go into or out of a tree item's toggle to expand/collapse it, as well as be able to select a tree item.

**Is this issue blocking you?**
N/A

**Screenshots**
N/A

**What is your product and what release date are you targeting?**
N/A

**Any other information?**
N/A
